java/flight/flight-sql-jdbc-core/src/main/java/org/apache/arrow/driver/jdbc/utils/AvaticaParameterBinder.java

```diff
@@ -0,0 +1,233 @@
+import org.apache.arrow.driver.jdbc.converter.impl.NullAvaticaParameterConverter;
+import org.apache.arrow.driver.jdbc.converter.impl.StructAvaticaParameterConverter;
+import org.apache.arrow.driver.jdbc.converter.impl.TimeAvaticaParameterConverter;
+import org.apache.arrow.driver.jdbc.converter.impl.TimestampAvaticaParameterConverter;
+import org.apache.arrow.driver.jdbc.converter.impl.UnionAvaticaParameterConverter;
+import org.apache.arrow.driver.jdbc.converter.impl.Utf8AvaticaParameterConverter;
+import org.apache.arrow.memory.BufferAllocator;
+import org.apache.arrow.vector.FieldVector;
+import org.apache.arrow.vector.VectorSchemaRoot;
+import org.apache.arrow.vector.types.pojo.ArrowType;
+import org.apache.calcite.avatica.remote.TypedValue;
+
+/**
+ * Convert Avatica PreparedStatement parameters from a list of TypedValue to Arrow and bind them to the
+ * VectorSchemaRoot representing the PreparedStatement parameters.
+ * <p>
+ * NOTE: Make sure to close the parameters VectorSchemaRoot once we're done with them.
+ */
+public class AvaticaParameterBinder {
+  private final PreparedStatement preparedStatement;
+  private final VectorSchemaRoot parameters;
+
+  public AvaticaParameterBinder(PreparedStatement preparedStatement, BufferAllocator bufferAllocator) {
+    this.parameters = VectorSchemaRoot.create(preparedStatement.getParameterSchema(), bufferAllocator);
+    this.preparedStatement = preparedStatement;
+  }
+
+  /**
+   * Bind the given Avatica values to the prepared statement.
+   * @param typedValues The parameter values.
+   */
+  public void bind(List<TypedValue> typedValues) {
+    bind(typedValues, 0);
+  }
+
+  /**
+   * Bind the given Avatica values to the prepared statement at the given index.
+   * @param typedValues The parameter values.
+   * @param index index for parameter.
+   */
+  public void bind(List<TypedValue> typedValues, int index) {
+    if (preparedStatement.getParameterSchema().getFields().size() != typedValues.size()) {
+      throw new IllegalStateException(
+          String.format("Prepared statement has %s parameters, but only received %s",
+              preparedStatement.getParameterSchema().getFields().size(),
+              typedValues.size()));
+    }
+
+    for (int i = 0; i < typedValues.size(); i++) {
+      bind(parameters.getVector(i), typedValues.get(i), index);
+    }
+
+    if (!typedValues.isEmpty()) {
+      parameters.setRowCount(index + 1);
+      preparedStatement.setParameters(parameters);
+    }
+  }
+
+  /**
+   * Bind a TypedValue to the given index on the FieldVctor.
+   *
+   * @param vector     FieldVector to bind to.
+   * @param typedValue TypedValue to bind to the vector.
+   * @param index      Vector index to bind the value at.
+   */
+  private void bind(FieldVector vector, TypedValue typedValue, int index) {
+    try {
+      if (!vector.getField().getType().accept(new BinderVisitor(vector, typedValue, index))) {
+        throw new RuntimeException(
```

Comment:
nit: is there a more appropriate exception than RuntimeException?

Reply:
I was thinking some type of `NotImplementedException` would be ideal but I couldn't find an exception like that. Any thoughts on an alternative? `UnsupportedOperationException` seem to imply that it will not be supported, which might not be what we want.

Reply:
IMO `UnsupportedOperationException` is fine for this sort of thing